### PR TITLE
Feature/api user

### LINKED ISSUE #1: Add user api
Create endpoint for user api

- [x] endpoint for GET
- [x] endpoint for POST

### LINKED ISSUE #2: Add event api
Create end points for event api
- [x] end point for GET

### LINKED ISSUE #3: Add patch for event api
Dev should be able to send PATCH request to event api end point

### PR DESCRIPTION
**Pull request by okalimbet**
## What is the change? 
Added ivolunteer data
Added get, post, and patch methods to manipulate and receive data
Added cors package
Added nodemon package

## What does it fix?
N/A

## Is this a fix or a feature? 
Feature

## Where should the reviewer start?
ivolunteerData.js line 1
server.js line 1

## How should this be tested?
By requesting info using get method to receive all events and user info
By using post method, user can add a new posting to their upcoming jobs
By using patch method, user updates the number of open jobs in the event

## Completed Issues
#1 #2 #3

### CODE REVIEW

File: ivolunteerData.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
+const iVolunteerData = {
+  users: [
+    {
```
Comment (coopterrones) [inline ivolunteerData.js:3]: User initial data and set up looks great.

File: ivolunteerData.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        {
+          id: '1-posting-23',
+          eventName: 'Help a local animal shelter',
+          positionName: 'dog walkers',
+          date: '2021/03/01'
+        }
+      ]
+    }
+  ],
+  events: [
```
Comment (coopterrones) [inline ivolunteerData.js:19]: Later we can hopefully make a goal of adding own postings

File: server.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  if (correctUser && registeredJob.id && eventName && positionName && date) {
+    correctUser.upcomingJobs.push(registeredJob);
+    response.status(200).json(`The ${correctUser.name} has registered for ${registeredJob.positionName}`)
+  } else {
+    response.status(200).json('Please make sure the user id and all position info are correct')
+  } 
+})
+
+app.patch('/events/:id', (request, response) => {
```
Comment (coopterrones) [inline server.js:48]: Patch working on our end

**Review by yiranli624 — APPROVED** [review]
Great work! Will need to deploy in the near future.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+const iVolunteerData = require('./ivolunteerData');
+
+app.set('port', process.env.PORT || 3001); 
+app.use(express.json());
+app.use(cors());
+app.locals.title = 'iVolunteer API';
+app.locals.users = iVolunteerData.users;
+app.locals.events = iVolunteerData.events;
+
+app.listen(app.get('port'), () => {
```
Comment (coopterrones) [inline server.js:13]: thumbs up

File: server.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  console.log(`${app.locals.title} is running on http://localhost:${app.get('port')}`);
+})
+
+app.get('/users/:id', (request, response) => {
```
Comment (coopterrones) [inline server.js:17]: get working!

File: server.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  const { id } = request.params;
+  const user = app.locals.users.find(user => user.id === parseInt(id))
+  response.json(user);
+})
+
+app.get('/events', (request, response) => {
```
Comment (coopterrones) [inline server.js:23]: working

File: server.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  response.json(app.locals.events);
+})
+
+app.get('/events/:id', (request, response) => {
```
Comment (coopterrones) [inline server.js:27]: looks great

File: server.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  const { id } = request.params;
+  const event = app.locals.events.find(event => event.id === id)
+
+  response.json(event);
+})
+
+app.post('/users/:id', (request, response) => {
```
Comment (coopterrones) [inline server.js:34]: post working on our end

**Pull request merged**